### PR TITLE
UI tests waitforfoldercreation

### PR DESCRIPTION
**Pull request by individual-it**
check if the folder was created

### CODE REVIEW

File: tests/ui/features/lib/FilesPage.php [hunk 1]
```diff
@@ -132,6 +137,30 @@ public function createFolder($name = null) {
 			// Used to work fine in 1.3.1 but now throws this exception
 			// Actually all that we need does happen, so we just don't do anything
 		}
+		$timeoutMsec = (int) $timeoutMsec;
+		$currentTime = microtime(true);
+		$end = $currentTime + ($timeoutMsec / 1000);
+		while ($currentTime <= $end) {
```
Comment (phil-davis) [inline tests/ui/features/lib/FilesPage.php]: Why is this outer loop needed?
We just want to do the content of this loop once:
a) loop until the newFolderButton has gone
b) loop until the folder is found in the list
sleeping and incrementing the time in each loop.

File: tests/ui/features/lib/FilesPage.php [hunk 2]
```diff
@@ -132,6 +137,30 @@ public function createFolder($name = null) {
...
+			while ($currentTime <= $end) {
```
Comment (phil-davis) [inline tests/ui/features/lib/FilesPage.php]: $currentTime never changes inside the loop - so the only way out is ``break``
If that never happens (or takes longer than the timeout) we will not get out.

File: tests/ui/features/lib/FilesPage.php [hunk 3]
```diff
@@ -132,6 +137,30 @@ public function createFolder($name = null) {
...
+				$newFolderButton = $this->find("xpath", $this->newFolderButtonXpath);
+				if ($newFolderButton === null || !$newFolderButton->isVisible()) {
+					break;
+				}
+			}
+			while ($currentTime <= $end) {
```
Comment (phil-davis) [inline tests/ui/features/lib/FilesPage.php]: same for this loop - if the folder never appears in the list, we will look forever, and get the job killed by Travis "no output after 10 minutes"

**Comment by individual-it** [comment]
@phil-davis  you  are right, changed it

**Comment by individual-it** [comment]
moved to oC original fork

**Pull request closed without merging**